### PR TITLE
Add SaslHandshakeRequest

### PR DESCRIPTION
**Pull request by codesnik**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PROJECT = kafka_protocol
 PROJECT_DESCRIPTION = Kafka protocol erlang library
-PROJECT_VERSION = 0.7.8
+PROJECT_VERSION = 0.7.9
```
Comment (zmstone) [inline Makefile]: 0.8

File: src/kpro.erl [hunk 2]
```diff
@@ -65,7 +65,7 @@
 -type int64()      :: -9223372036854775808..9223372036854775807.
 -type str()        :: undefined | string() | binary().
 -type bytes()      :: undefined | binary().
--type api_key()    :: 0..16.
+-type api_key()    :: 0..17.
```
Comment (zmstone) [inline src/kpro.erl]: 0.8.0

**Comment by zmstone** [comment]
Thanks a lot!
Let's make it 0.8

File: src/kafka_protocol.app.src [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {application,kafka_protocol,
              [{description,"Kafka protocol erlang library"},
-              {vsn,"0.7.8"},
+              {vsn,"0.7.9"},
```
Comment (id) [inline src/kafka_protocol.app.src]: 0.8.0

File: priv/kafka.bnf [hunk 4]
```diff
@@ -250,3 +250,9 @@ DescribeGroupsResponse => [GroupDescription]
     MemberMetadata => bytes
     MemberAssignment => bytes
 
+SaslHandshakeRequest => Mechanism
+  Mechanism => string
+
+SaslHandshakeResponse => ErrorCode [EnabledMechanism]
+  ErrorCode => int16
+  EnabledMechanism => string
```
Comment (id) [inline priv/kafka.bnf]: It's "enabled mechanisms" - "Array of mechanisms enabled in the server". See SaslHandshake API in https://kafka.apache.org/protocol.html.

Comment (codesnik) [reply priv/kafka.bnf]: but `[EnabledMechanism]` *is* an array of `enabled mechanism`s. I just tried to blend into the current code style.
Looks like it'd be the first plural in the whole file. Naming conventions probably could be changed in one pass (looks like your https://github.com/klarna/kafka_protocol/pull/21 does exactly that), but, ok, I'm changing it.

Comment (id) [reply priv/kafka.bnf]: You are right, I think it's a typo or a mistake in kafka protocol specification.

Please don't change it!

Comment (zmstone) [reply priv/kafka.bnf]: Yeah, keep it for now. This naming convention is derived from  kafka 0.8 spec. 
We'll change naming convention in 0.9 or maybe make it 1.0

**Review by zmstone — APPROVED** [review]

**Review by id — APPROVED** [review]
👍

**Comment by codesnik** [comment]
> You are right, I think it's a typo or a mistake in kafka protocol
> specification.
>
> Please don't change it!
>
Oopsie.
-- 
Alexey Trofimenko

**Pull request merged**